src/mame/roland/roland_d70.cpp

```diff
@@ -0,0 +1,562 @@
+  PORT_BIT(0x8, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("F2") PORT_CODE(KEYCODE_S)
+  PORT_BIT(0x4, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("F1") PORT_CODE(KEYCODE_A)
+  PORT_BIT(0x2, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("User") PORT_CODE(KEYCODE_STOP)
+  PORT_BIT(0x1, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Part") PORT_CODE(KEYCODE_COMMA)
+
+  PORT_START("KEY5")
+  PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Edit") PORT_CODE(KEYCODE_X)
+  PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Portamento")
+  PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Resonance")
+  PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Pan")
+  PORT_BIT(0x8, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Tuning") PORT_CODE(KEYCODE_V)
+  PORT_BIT(0x4, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Attack")
+  PORT_BIT(0x2, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Release")
+  PORT_BIT(0x1, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Pcm Card") PORT_CODE(KEYCODE_C)
+
+  PORT_START("KEY6")
+  PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Play") PORT_CODE(KEYCODE_Z)
+  PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Solo")
+  PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Cutoff")
+  PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Level")
+  PORT_BIT(0x8, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Upper 4")
+  PORT_BIT(0x4, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Upper 3")
+  PORT_BIT(0x2, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Lower 2")
+  PORT_BIT(0x1, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Lower 1")
+
+  PORT_START("KEY7")
+  PORT_BIT(0x80, IP_ACTIVE_LOW, IPT_UNUSED)
+  PORT_BIT(0x40, IP_ACTIVE_LOW, IPT_UNUSED)
+  PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_UNUSED)
+  PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_UNUSED)
+  PORT_BIT(0x8, IP_ACTIVE_LOW, IPT_UNUSED)
+  PORT_BIT(0x4, IP_ACTIVE_LOW, IPT_UNUSED)
+  PORT_BIT(0x2, IP_ACTIVE_LOW, IPT_UNUSED)
+  PORT_BIT(0x1, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Effect/Ctrl") PORT_CODE(KEYCODE_M)
+INPUT_PORTS_END
+
+class roland_d70_state : public driver_device {
+public:
+  roland_d70_state(const machine_config &mconfig, device_type type,
+                   const char *tag)
+      : driver_device(mconfig, type, tag), cpu(*this, "maincpu"),
+        pcm(*this, "pcm"), lcd(*this, "lcd"), midi_timer(*this, "midi_timer"),
+        ram(*this, "ram"), dsp_ram(*this, "dsp_ram"), m_keys(*this, "KEY%u", 0),
+        sw_scan_index(0), sw_scan_bank(-1), sw_scan_state(0xFF), sw_scan_mode(0),
+        sw_scan_current_out(0xFF), m_midi_rx(0), m_midi_pos(0) {}
+
+  void d70(machine_config &config);
+
+  void init_d70();
+
+protected:
+  virtual void machine_start() override;
+  virtual void machine_reset() override;
+
+private:
+  required_device<i8x9x_device> cpu;
+  required_device<mb87419_mb87420_device> pcm;
+  required_device<t6963c_device> lcd;
+  required_device<timer_device> midi_timer;
+  required_device<ram_device> ram;
+  required_device<ram_device> dsp_ram;
+  required_ioport_array<8> m_keys;
```

Comment:
We prefer but don't require the m_ prefix for member variables.  Consistency is more important though, and I'm flagging this because having one m_ and the rest not stands out a bit.